x-pack/plugins/apm/public/components/shared/charts/duration_distribution_chart/index.tsx

```diff
@@ -266,26 +269,28 @@ export function DurationDistributionChart({
             ticks={yTicks}
             gridLine={{ visible: true }}
           />
-          {data.map((d, i) => (
+          {data.map((d) => (
             <AreaSeries
               key={d.id}
               id={d.id}
               xScaleType={ScaleType.Log}
               yScaleType={ScaleType.Log}
-              data={replaceHistogramDotsWithBars(d.histogram)}
+              data={replaceHistogramZerosWithMinimumDomainValue(d.histogram)}
```

Comment:
Do you think wrapping it on a `useMemo` would help? At least it wouldn't recalculate on every render.

Reply:
👍 Moved this to a `useMemo()` in [52cf77e](https://github.com/elastic/kibana/pull/138259/commits/52cf77ecea2b6a95a901af721beae8c97d5bcf4c).